#!/usr/bin/env python
PACKAGE = "cbot_control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("yaw_k",    double_t,    0, "LQR yaw gain K", 2.8 ,  0.001, 200.0)
gen.add("yaw_kr",    double_t,    0, "LQR yaw gain Kr", 1.8 ,  0.001, 200.0)
gen.add("yaw_ki",    double_t,    0, "LQR yaw gain Ki", 0.7 ,  0.001, 200.0)
gen.add("yaw_kaw",    double_t,    0, "LQR yaw gain kaw", 10.0 ,  0.001, 200.0)
gen.add("yaw_dm_saturation",    double_t,    0, "Pitch differential mode saturation for thrusters", 18.0 ,  0.01, 18.0)

gen.add("pitch_k",    double_t,    0, "LQR pitch gain K", 0.7305 ,  0.001, 200.0)
gen.add("pitch_kr",    double_t,    0, "LQR pitch gain Kr", 0.5451 ,  0.001, 200.0)
gen.add("pitch_ki",    double_t,    0, "LQR pitch gain Ki", 0.5 ,  0.001, 200.0)
gen.add("pitch_kaw",    double_t,    0, "LQR pitch gain kaw", 10.0 ,  0.001, 200.0)
gen.add("pitch_dm_saturation",    double_t,    0, "Pitch differential mode saturation for thrusters", 10.0 ,  0.01, 10.0)

gen.add("depth_k",    double_t,    0, "Depth proportional gain K", 16 ,  0.001, 200.0)
gen.add("depth_ki",    double_t,    0, "Depth integral gain Ki", 0.5 ,  0.001, 200.0)	
gen.add("depth_cm_saturation",    double_t,    0, "Depth common mode saturation for thrusters", 20.0 ,  0.01, 20.0)

gen.add("speed_k",    double_t,    0, "Velocity proportional gain K", 40 ,  0.001, 200.0)
gen.add("speed_ki",    double_t,    0, "Velocity integral gain Ki", 20,  0.001, 200.0)
gen.add("speed_cm_saturation",    double_t,    0, "Speed common mode saturation for thrusters", 60.0 ,  0.01, 60.0)

gen.add("controller_on",    int_t,    0, "Controlller swith", 0 ,  0, 1)
gen.add("heading_ctrl",    int_t,    0, "Heading control swith", 0 ,  0, 1)
gen.add("pitch_ctrl",    int_t,    0, "Pitch control swith", 0 ,  0, 1)
gen.add("speed_ctrl",    int_t,    0, "Speed control swith", 0 ,  0, 1)
gen.add("depth_ctrl",    int_t,    0, "Depth control swith", 0 ,  0, 1)
gen.add("roll_ctrl",    int_t,    0, "Roll control swith", 0 ,  0, 1)

exit(gen.generate(PACKAGE, "Config", "Controllers"))
